gocov/convert/convert.go

```diff
@@ -74,10 +58,35 @@ func ConvertProfiles(filenames ...string) ([]byte, error) {
 		if err != nil {
 			return nil, err
 		}
-		for _, p := range profiles {
-			if err := converter.convertProfile(packages, localModuleName, p); err != nil {
+
+		sort.Slice(profiles, func(i, j int) bool {
+			return profiles[i].FileName < profiles[j].FileName
+		})
+
+		//packageNameGoPkgsMap := make(map[string][]*goPackages.Package)
+		//profilePkgsMap := make(map[*cover.Profile][]*goPackages.Package)
+		tempCache := make(map[string][]*goPackages.Package)
+		var prevPackage string
+		for _, profile := range profiles {
+			packageName := filepath.Dir(profile.FileName)
+
+			var pkgs []*goPackages.Package
+			if packageName != prevPackage {
+				delete(tempCache, prevPackage)
+				pkgs, err = goPackages.Load(&goPackages.Config{
+					Mode: goPackages.NeedName | goPackages.NeedSyntax | goPackages.NeedTypes | goPackages.NeedCompiledGoFiles,
```

Comment:
I don't think types, syntax, or compiled go files are needed. We should just need the name & module.